libraries/ESP8266mDNS/src/LEAmDNS_Priv.h

```diff
@@ -0,0 +1,168 @@
+#endif
+
+// Enable/disable debug trace macros
+#ifdef DEBUG_ESP_MDNS_RESPONDER
+//#define DEBUG_ESP_MDNS_INFO
+#define DEBUG_ESP_MDNS_ERR
+#define DEBUG_ESP_MDNS_TX
+#define DEBUG_ESP_MDNS_RX
+#endif
+
+#ifdef DEBUG_ESP_MDNS_RESPONDER
+    #ifdef DEBUG_ESP_MDNS_INFO
+        #define DEBUG_EX_INFO(A)    A
+    #else
+        #define DEBUG_EX_INFO(A)
+    #endif
+    #ifdef DEBUG_ESP_MDNS_ERR
+        #define DEBUG_EX_ERR(A) A
+    #else
+        #define DEBUG_EX_ERR(A)
+    #endif
+    #ifdef DEBUG_ESP_MDNS_TX
+        #define DEBUG_EX_TX(A)  A
+    #else
+        #define DEBUG_EX_TX(A)
+    #endif
+    #ifdef DEBUG_ESP_MDNS_RX
+        #define DEBUG_EX_RX(A)  A
+    #else
+        #define DEBUG_EX_RX(A)
+    #endif
+
+    #ifdef DEBUG_ESP_PORT
+        #define DEBUG_OUTPUT DEBUG_ESP_PORT
+    #else
+        #define DEBUG_OUTPUT Serial
+    #endif
+#else
+    #define DEBUG_EX_INFO(A)
+    #define DEBUG_EX_ERR(A)
+    #define DEBUG_EX_TX(A)
+    #define DEBUG_EX_RX(A)
+#endif
+
+
+/*  Replaced by 'lwip/prot/dns.h' definitions
+#ifdef MDNS_IP4_SUPPORT
+    #define MDNS_MULTICAST_ADDR_IP4     (IPAddress(224, 0, 0, 251)) // ip_addr_t v4group = DNS_MQUERY_IPV4_GROUP_INIT
+#endif
+#ifdef MDNS_IP6_SUPPORT
+    #define MDNS_MULTICAST_ADDR_IP6     (IPAddress("FF02::FB"))     // ip_addr_t v6group = DNS_MQUERY_IPV6_GROUP_INIT
+#endif*/
+//#define MDNS_MULTICAST_PORT               5353
+
+/*
+ * This is NOT the TTL (Time-To-Live) for MDNS records, but the
+ * subnet level distance MDNS records should travel.
+ * 1 sets the subnet distance to 'local', which is default for MDNS.
+ * (Btw.: 255 would set it to 'as far as possible' -> internet)
+ */
+#define MDNS_MULTICAST_TTL              1
+
+/*
+ * This is the MDNS record TTL
+ * Host level records are set to 2min (120s)
+ * service level records are set to 75min (4500s)
+ */
+#define MDNS_HOST_TTL                   120
+#define MDNS_SERVICE_TTL                4500
+
+/*
+ * Compressed labels are flaged by the two topmost bits of the length byte being set
+ */
+#define MDNS_DOMAIN_COMPRESS_MARK       0xC0
+/*
+ * Avoid endless recursion because of malformed compressed labels
+ */
+#define MDNS_DOMAIN_MAX_REDIRCTION      6
+
+/*
+ * Default service priority and weight in SRV answers
+ */
+#define MDNS_SRV_PRIORITY               0
+#define MDNS_SRV_WEIGHT                 0
+
+/*
+ * Delay between and number of probes for host and service domains
+ */
+#define MDNS_PROBE_DELAY                250
+#define MDNS_PROBE_COUNT                3
+
+
+/*
+ * Force host domain to use only lowercase letters
+ */
+//#define MDNS_FORCE_LOWERCASE_HOSTNAME
+
+/*
+ * Enable/disable the usage of the F() macro in debug trace printf calls.
+ * There needs to be an PGM comptible printf function to use this.
+ *
+ * USE_PGM_PRINTF and F
+ */
+#define USE_PGM_PRINTF
+
+#ifdef USE_PGM_PRINTF
+// See: https://github.com/esp8266/Arduino/issues/3369
+#define PROGMEM_LEA __attribute__((section(".irom.text.LEA")))

```

Comment:
PROGMEM variables are now uniquified, is this still required?

Reply:
I‘ve got linkage problems without this...

Reply:
I think the unique sections haven't been released yet and are only available in latest git. I also suspect that @LaborEtArs is really working on a hand-modified 2.4.2 code base, which means the unique sections aren't available in his build.